day4/scratchcards.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Calculates the total sum of all winnings from the scratch cards
+ * @param {string[]} data The raw data provided in the problem
+ * @returns {int} The actual winnings
+ */
+export function calculateWinnings(data) {
+  return data
+    .map((line) => {
+      let matches = -1;
+
+      const lineSplit = line.split("|");
+      const winningNumbers = lineSplit[0]
+        .split(":")[1]
+        .trim()
+        .split(" ")
+        .map((s) => parseInt(s))
+        .filter((n) => !isNaN(n));
+
+      const foundNumbers = lineSplit[1]
+        .trim()
+        .split(" ")
+        .map((s) => parseInt(s))
+        .filter((n) => !isNaN(n));
+
+      console.log(foundNumbers);
+
+      for (const num of foundNumbers) {
+        if (winningNumbers.find((n) => n === num)) {
+          matches++;
+        }
+      }
```

Comment:
So yeah... This is definitely not the most performant way to do this. We're creating lots of unnecessary memory, and doing tons of unnecessary copying. If I were going for performance, I probably would do the parsing first to create the two number arrays, then I'd do a sort, followed by a better search algorithm.